CI/TST: Enable -W error:::pandas in pyproject.toml

mroeschke opened the pull request:
- [ ] closes #48553 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

mroeschke committed:
CI/TST: Enable -W error:::pandas in pyproject.toml

mroeschke committed:
Address some tests

mroeschke committed:
Merge remote-tracking branch 'upstream/main' into tst/warnings_errors_pandas

mroeschke committed:
filter more expected warnings in tests

mroeschke committed:
Change examples raising warnings

mroeschke committed:
Merge remote-tracking branch 'upstream/main' into tst/warnings_errors_pandas

mroeschke committed:
Just ignore userwarning, not point of the test

mroeschke committed:
remove a sql date example, ignore open plots

mroeschke committed:
Merge remote-tracking branch 'upstream/main' into tst/warnings_errors_pandas

mroeschke committed:
Just use one example

mroeschke committed:
Fix path

mroeschke committed:
Merge remote-tracking branch 'upstream/main' into tst/warnings_errors_pandas

jbrockmendel commented on pandas/tests/series/test_arithmetic.py:
```diff
@@ -349,6 +349,7 @@ def test_add_list_to_masked_array(self, val, dtype):
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.filterwarnings("ignore:evaluating in Python space:UserWarning")
     def test_add_list_to_masked_array_boolean(self):
```
side note: ive seen these and wondered why the lists aren't getting wrapped before being passed to the EA method

mroeschke replied:
I'll add a todo comment here to that effect

mroeschke committed:
Merge remote-tracking branch 'upstream/main' into tst/warnings_errors_pandas

mroeschke committed:
Add comment about filtering warning

mroeschke commented on pandas/io/sql.py:
```diff
@@ -632,36 +632,9 @@ def read_sql(
 
     Apply date parsing to columns through the ``parse_dates`` argument
-
-    >>> pd.read_sql('SELECT int_column, date_column FROM test_data',
```
I removed these examples because they trigger UserWarnings that the format could not be inferred. There is still an example of `parse_dates` at the bottom that I think fits the spirit of showing what the keyword does

mroeschke committed:
Merge remote-tracking branch 'upstream/main' into tst/warnings_errors_pandas

mroeschke committed:
Filter out expected warning?

mroeschke committed:
Merge remote-tracking branch 'upstream/main' into tst/warnings_errors_pandas

mroeschke committed:
Merge remote-tracking branch 'upstream/main' into tst/warnings_errors_pandas

mroeschke committed:
Remove unneeded filter

mroeschke committed:
Merge remote-tracking branch 'upstream/main' into tst/warnings_errors_pandas

jbrockmendel commented on pyproject.toml:
```diff
@@ -395,6 +395,9 @@ doctest_optionflags = [
   "ELLIPSIS",
 ]
 filterwarnings = [
+  "error:::pandas",
```
what does this one mean?

mroeschke replied:
If any unhanded warning originates from the path `pandas/*`, treat it as as an error in the test suite.

Any subsequent filter in this list takes prescient over this one if they overlap.

phofl commented:
needs a rebase

mroeschke committed:
Merge remote-tracking branch 'upstream/main' into tst/warnings_errors_pandas

mroeschke committed:
Try catching sockerwarning in pyproject toml

mroeschke committed:
Add back sslsocket filtering

mroeschke committed:
add back matplotlib error too

mroeschke commented:
Rebased. Added back the `ResourceWarning` check I removed in https://github.com/pandas-dev/pandas/pull/51443. Appears `tm.assert_produces_warning` still catches these

mroeschke committed:
Merge remote-tracking branch 'upstream/main' into tst/warnings_errors_pandas

mroeschke committed:
include patrick's changes

mroeschke committed:
Typo

phofl commented:
thx @mroeschke
